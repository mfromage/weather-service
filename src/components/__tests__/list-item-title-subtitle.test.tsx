import ListItemTitleSubtitle from '@/components/list-item-title-subtitle';
import {fireEvent, render} from '@testing-library/react-native';

describe('ListItemTitleSubtitle', () => {
  const setup = (props = {}) => {
    const testId = 'list-item-title-subtitle-test-id';
    const defaultProps = {
      title: 'Hello',
      subtitle: 'world',
      item: {id: testId},
      onPress: jest.fn(),
    };
    const utils = render(
      <ListItemTitleSubtitle {...defaultProps} {...props} testID={testId} />,
    );
    const listItemTitleSubtitle = utils.getByTestId(testId);
    return {...utils, listItemTitleSubtitle};
  };

  it('render with given props', () => {
    const title = 'This is title';
    const subtitle = 'not a title';

    const {getByText} = setup({title, subtitle});
    expect(getByText(title)).toBeTruthy();
    expect(getByText(subtitle)).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const {toJSON} = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('call onPress with given item', () => {
    const onPress = jest.fn();
    const item = {id: 1, name: 'hello'};
    const {listItemTitleSubtitle} = setup({item, onPress});

    fireEvent.press(listItemTitleSubtitle);
    expect(onPress).toHaveBeenCalledWith(item);
  });
});
