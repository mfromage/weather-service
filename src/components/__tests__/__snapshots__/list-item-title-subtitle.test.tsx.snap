// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemTitleSubtitle matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="list-item-title-subtitle-test-id"
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderBottomWidth": 1,
        "borderColor": "lightgray",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 12,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Hello
    </Text>
    <Text
      style={
        {
          "fontSize": 12,
        }
      }
    >
      world
    </Text>
  </View>
</View>
`;
